### ISSUE TITLE
limit number of retries for wireless connect and do exit 0 if unsuccessful

### ISSUE THREAD

**Issue by vdelendik** [zebrunner/appium#360]
as of now we try to connect for hours. let's make 3-5min timeout and do exit 0 if no way to do. 3-5m should cover avg reboot cycle for phone/tablet/tv

```
Connecting to: 10.151.206.23:5555
03-15 13:46:34.966   101  3845 W adb     : network.cpp:149 failed to connect to '10.151.206.23:5555': No route to host
failed to connect to '10.151.206.23:5555': No route to host
Connecting to: 10.151.206.23:5555
03-15 13:46:42.976   101  3859 W adb     : network.cpp:149 failed to connect to '10.151.206.23:5555': No route to host
failed to connect to '10.151.206.23:5555': No route to host
Connecting to: 10.151.206.23:5555
03-15 13:46:51.029   101  3873 W adb     : network.cpp:149 failed to connect to '10.151.206.23:5555': No route to host
failed to connect to '10.151.206.23:5555': No route to host

```

**Comment by vdelendik** [comment]
number of retries limited b y10 with 5 sec interval by default:
https://github.com/zebrunner/appium/blob/1c73e8e5887d19ff7f506fff745e34fe2d3076b7/device_connect.sh#L57

changes are available for testing in `appium:2.0.14` because I already applied changes on one of the prod as a hot fix

**Comment by vdelendik** [comment]
successfully verified on inaccessible wireless device:
```
03-15 16:52:24.292    84   208 W adb     : network.cpp:149 failed to connect to '10.146.84.44:5555': No route to host
failed to connect to '10.146.84.44:5555': No route to host
Device 10.146.84.44:5555 is not connected!
Exiting without restarting...
```
